inference-engine/src/vpu/common/src/ngraph/transformations/merge_subsequent_dsr_operations.cpp

```diff
@@ -20,8 +20,15 @@ MergeSubsequentDSROperations::MergeSubsequentDSROperations() {
         if (!predecessor) {
             return false;
         }
-
-        dsr->input(0).replace_source_output(predecessor->input_value(0));
+        // this will create a new DSR with correct inputs
+        auto newDsr = dsr->copy_with_new_inputs({predecessor->input_value(0), dsr->input_value(1)});
+        // replace DSR2 with new so DSR2 will lose all consumers so it will die after pass execution
+        replace_node(dsr, newDsr);
+        // reconnect all DSR1 consumers even with DSR2 which will be destructed so this is no more an issue
+        for (auto &consumer : predecessor->get_output_target_inputs(0)) {
+            consumer.replace_source_output(newDsr);
+        }
+        newDsr->input(0).replace_source_output(predecessor->input_value(0));
```

Comment:
You don't need this as you already created newDsr with `predecessor->input_value(0)` input.